src/unitxt/operators.py

```diff
@@ -644,20 +651,96 @@ def process(self, instance: Dict[str, Any], stream_name: str = None) -> Dict[str
 
 class FilterByValues(SingleStreamOperator):
     """
-    Filters a stream, yielding only instances that match specified values.
+    Filters a stream, yielding only instances that match specified values in the provided fields.
 
     Args:
-        values (Dict[str, Any]): The values that instances should match to be included in the output.
+        values (Dict[str, Any]): For each field, the values that instances should match to be included in the output.
     """
 
-    values: Dict[str, Any]
+    required_values: Dict[str, Any]
+
+    def process(self, stream: Stream, stream_name: str = None) -> Generator:
+        for instance in stream:
+            filter = False
+            for key, value in self.required_values.items():
+                if not key in instance:
+                    raise ValueError(f"Required filter field ('{key}') in FilterByValues is not found in {instance}")
+                if instance[key] != value:
+                    filter = True
```

Comment:
`break` at this point? or is the intention to keep the loop so all the keys are checked for their presence?

Reply:
I did not think about it originally, but I indeed think it's best to see the keys are in the instance.